packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -210,4 +217,70 @@ export class MetadataOrchestrator {
         return 'string';
     }
   };
+
+  public eligibilityDelegate = async (
+    requests: ApexOASEligiblePayload,
+    maxRetries: number = 3
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    let response = {};
+    let attempt = 0;
+    const languageClient = languageClientUtils.getClientInstance();
+    if (languageClient) {
+      while (attempt < maxRetries) {
+        try {
+          attempt++;
+          response = await languageClient?.sendRequest('apexoas/isEligible', requests);
+          telemetryService.sendEventData('isEligibleResponseSucceeded', {
+            classNumbers: requests.payload.length.toString()
+          });
+          break;
+        } catch (error) {
+          telemetryService.sendException('isEligibleResponseFailed', `${error} after trying ${attempt} times.`);
+
+          if (attempt >= maxRetries) {
+            // fallback TBD after we understand it better
+            throw new Error(nls.localize('cannot_get_apexoaseligibility_response'));
+          }
+        }
+      }
+    }
+    return response as ApexClassOASEligibleResponses;
+  };
+
+  public validateEligibility = async (
+    sourceUri: vscode.Uri | vscode.Uri[],
+    isMethodSelected: boolean = false
+  ): Promise<ApexClassOASEligibleResponses> => {
+    const telemetryService = await getTelemetryService();
+    if (Array.isArray(sourceUri)) {
+      throw new Error('We do not consider list of src files now');
+    }
+    let cursorPosition;
+    if (isMethodSelected) {
+      const editor = vscode.window.activeTextEditor;
+      if (editor && this.getFileExtension(editor.document) === 'cls') {
+        cursorPosition = editor.selection.active;
+      } else {
+        telemetryService.sendException('activeTextEditorNotApex', nls.localize('active_text_editor_not_apex'));
+        throw new Error(nls.localize('invalid_active_text_editor'));
+      }
+    }
+    // generate the payload
+    const request: ApexClassOASEligibleRequest = {
+      resourceUri: sourceUri.path,
+      includeAllMethods: !isMethodSelected,
+      includeAllProperties: !isMethodSelected,
+      positions: cursorPosition ? [cursorPosition] : null,
```

Comment:
This must be an array or null, since all requests are going to the same endpoint. Java is strongly typed and does not allow union types. The other choice would be to change the position to only a single value. Let's discuss further.

Reply:
Got it. I still cannot figure out what use case leads to multiple cursor positions, since there is only one cursor.

Reply:
@mingxuanzhangsfdx I am on the fence with this too.